File: Comparing Strings/Comparing Strings.py

string1 = input()
string2 = input()
#result = True
if len(string1) != len(string2):
    print("NO")
   
else:
   chars_to_be_swap = []
   for char in range(len(string1)):
        if string1[char] != string2[char]:
            chars_to_be_swap.append(char)
    
    #print(chars_to_be_swap)
if len(chars_to_be_swap) != 2:
    print("NO")
    
else:
        
    string1 = list(string1)
    string1[chars_to_be_swap[0]], string1[chars_to_be_swap[1]] = string1[chars_to_be_swap[1]], string1[chars_to_be_swap[0]]
        
    string3 = "".join(string1) 

    if string3 == string2:
       print("YES")
    else:
       print("NO")
    
    
     


   
            
    

File: Comparing Strings/Comparing Strings.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main()
{   
    string string1,string2;
    
    cin >> string1 >> string2;
 
    if(string1.length() != string2.length())
    {
        cout <<"NO" << endl;
        return 0;
    }
 
    vector<int> swaping_elements;
    
    for(int i=0; i<string1.length(); i++)
    {
        if(string1[i] != string2[i])
        {
            swaping_elements.push_back(i);
        }
    }
    if(swaping_elements.size() != 2)
    {
        cout << "NO" << endl;
        return 0;
    }
 
    swap(string1[swaping_elements[0]], string1[swaping_elements[1]]);
 
    if(string1 == string2)
    {
        cout<<"YES"<<endl;
    }
    else
    {
        cout<<"NO"<<endl;
    }
    return 0;
    
}